index.js

```diff
@@ -236,6 +237,122 @@ Peer.prototype.send = function (chunk) {
   self._channel.send(chunk)
 }
 
+/**
+ * Add all tracks within a MediaStream to the connection.
+ * @param {MediaStream} stream
+ * @returns {RTCRtpSender[]}
+ */
+Peer.prototype.addStream = function (stream) {
+  var self = this
+
+  self._debug('addStream()')
+
+  var senders = []
+  stream.getTracks().forEach(function (track) {
+    senders.push(self.addTrack(track, stream))
+  })
+  return senders
+}
+
+/**
+ * Add a MediaStreamTrack to the connection. Returns the RTCRtpSender used to remove the track.
+ * @param {MediaStreamTrack} track
+ * @returns {RTCRtpSender}
+ */
+Peer.prototype.addTrack = function (track, stream) {
+  var self = this
+
+  self._debug('addTrack()')
+
+  return self._pc.addTrack(track, stream)
+}
+
+/**
+ * Add all tracks within a MediaStream to the connection.
+ * @param {MediaStream} stream
+ * @returns {RTCRtpSender[]}
+ */
+Peer.prototype.addStream = function (stream) {
+  var self = this
+
+  self._debug('addStream()')
+
+  var senders = []
+  stream.getTracks().forEach(function (track) {
+    senders.push(self.addTrack(track, stream))
+  })
+  return senders
+}
+
+/**
+ * Remove a RTCRtpSender from the connection.
+ * @param {RTCRtpSender} sender
+ * @returns {RTCRtpSender}
+ */
+Peer.prototype.removeSender = function (sender) {
+  var self = this
+
+  self._debug('removeSender()')
+
+  try {
+    self._pc.removeTrack(sender)
+  } catch (err) {
+    if (err.name === 'NS_ERROR_UNEXPECTED') {
+      self._sendersAwaitingStable.push(sender) // HACK: Firefox must wait until (signalingState === stable) https://bugzilla.mozilla.org/show_bug.cgi?id=1133874
+    } else {
+      self.destroy(err)
+    }
+  }
+}
+Peer.prototype.removeTrack = Peer.prototype.removeSender // for symmetry in the API
+
+/**
+ * Remove an array of RTCRtpSenders from the connection.
+ * @param {RTCRtpSender[]} sender
+ * @param {MediaStream} stream
+ * @returns {RTCRtpSender}
+ */
+Peer.prototype.removeSenders = function (senders) {
+  var self = this
+
+  self._debug('removeSenders()')
+
+  senders.forEach(function (sender) {
+    self.removeSender(sender)
+  })
+}
+Peer.prototype.removeStream = Peer.prototype.removeSenders // for symmetry in the API
+
+/**
+ * Renegotiate the peer connection.
+ */
+Peer.prototype.renegotiate = function () {
```

Comment:
I exposed it to allow batching of asynchronous changes to the connection. Typically MediaStreams come from several async sources in parallel (ie one from getUserMedia, one from screenCapture, a synchronous WebAudio input).

The API benefits of abstracting negotiation is something to consider though.

Reply:
If user expects all of these to be added with one negotiation, it is still possible to do `Promise.all()` and then add necessary streams/tracks synchronously so that only one renegotiation is made.